xapian-letor/ranker/ranker.cc

```diff
@@ -354,9 +354,6 @@ Ranker::labelcomparer(const FeatureVector & firstfv, const FeatureVector& second
 std::vector<Xapian::docid>
 Ranker::rank(const Xapian::MSet & mset, const string & model_key, const Xapian::FeatureList & flist)
 {
-    if (mset.empty()) {
-	throw Xapian::InvalidArgumentError("Empty MSet. No documents could be retrieved with the given Query.\n");
-    }
```

Comment:
I thought you said an empty `MSet` would cause problems, but you've just removed the check entirely rather than returning an empty vector here as I suggested.